File: @.PTM.℗™/^.tpm.fuzba=c&py+4/c+x_tpm_FUZPET_10]SHIP/c+x_tpm_FUZPET_6]PURE/plugins/increment_turn.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_LINE 256

int main() {
    // Read the current state file into a temporary buffer
    FILE *infile = fopen("pieces/clock/state.txt", "r");
    if (!infile) {
        // Create the directory if it doesn't exist
        system("mkdir -p pieces/clock");
        FILE *newfile = fopen("pieces/clock/state.txt", "w");
        if (newfile) {
            fprintf(newfile, "turn 0\ntime 08:00:00\n");
            fclose(newfile);
        }
        infile = fopen("pieces/clock/state.txt", "r");
        if (!infile) {
            return 1;
        }
    }
    
    char temp_lines[100][MAX_LINE];
    int line_count = 0;
    
    // Read all lines
    while (line_count < 100 && fgets(temp_lines[line_count], MAX_LINE, infile)) {
        line_count++;
    }
    fclose(infile);
    
    // Look for turn line and update it
    int turn_updated = 0;
    for (int i = 0; i < line_count; i++) {
        if (strncmp(temp_lines[i], "turn ", 5) == 0) {
            // Extract current turn
            int current_turn = 0;
            // Find the turn portion after "turn "
            char *turn_start = temp_lines[i] + 5; // Skip "turn "
            while (*turn_start == ' ') turn_start++; // Skip spaces
            // Parse the number
            current_turn = atoi(turn_start);
            
            // Increment turn
            current_turn++;
            
            // Update the line
            snprintf(temp_lines[i], MAX_LINE, "turn %d\n", current_turn);
            turn_updated = 1;
        }
    }
    
    // If no turn line was found, add one
    if (!turn_updated && line_count < 100) {
        strcpy(temp_lines[line_count], "turn 1\n");
        line_count++;
    }
    
    // Write the updated content back to the file
    FILE *outfile = fopen("pieces/clock/state.txt", "w");
    if (!outfile) {
        return 1;
    }
    
    for (int i = 0; i < line_count; i++) {
        fputs(temp_lines[i], outfile);
    }
    
    fclose(outfile);
    return 0;
}

File: @.PTM.℗™/pmo-gh/!.PMO_py_00.01/pieces/apps/fuzzpet_app/clock/plugins/increment_turn.py
#!/usr/bin/env python3
"""
increment_turn.py - Increments turn in pieces/clock/state.txt by 1
Based on the proven C implementation design
"""

import os
import sys

def main():
    # Ensure clock piece directory exists
    os.makedirs("pieces/clock", exist_ok=True)
    
    # Read the current state file
    state_file = "pieces/clock/state.txt"
    
    # Initialize with default state if file doesn't exist
    if not os.path.exists(state_file):
        with open(state_file, "w") as f:
            f.write("turn 0\ntime 08:00:00\n")
    
    with open(state_file, "r") as f:
        lines = f.readlines()
    
    # Process lines to find and update turn
    updated_lines = []
    turn_found = False
    
    for line in lines:
        if line.startswith("turn "):
            # Extract current turn
            current_turn_str = line[5:].strip()  # Skip "turn "
            
            try:
                current_turn = int(current_turn_str)
            except ValueError:
                # If parsing fails, use default
                current_turn = 0
            
            # Increment turn
            new_turn = current_turn + 1
            
            # Update the line
            updated_lines.append(f"turn {new_turn}\n")
            turn_updated = True
        else:
            updated_lines.append(line)
    
    # If no turn line was found, add one
    if not turn_updated:
        updated_lines.append(f"turn 1\n")
    
    # Write the updated content back to the file
    with open(state_file, "w") as f:
        f.writelines(updated_lines)

if __name__ == "__main__":
    main()